src/core/crypto/hdKey.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { hmac } from "@noble/hashes/hmac";
+import { sha512 } from "@noble/hashes/sha512";
+import * as bip39 from "@scure/bip39";
+
+export type DerivedKeys = {
+  key: Uint8Array;
+  chainCode: Uint8Array;
+};
+
+/**
+ * Aptos derive path is 637
+ */
+export const APTOS_HARDENED_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'?$/;
+export const APTOS_BIP32_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+\/[0-9]+$/;
```

Comment:
This is a BIP-44 regex (see https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#path-levels)